src/core/validation/constraints/unique.text.id.ts

```diff
@@ -25,7 +27,13 @@ export enum TextIdType {
 export class IsUniqueTextIdConstraint implements ValidatorConstraintInterface {
   async validate(textId: any, args: ValidationArguments): Promise<boolean> {
     const [target] = args.constraints;
-    if (target === TextIdType.challenge) {
```

Comment:
This is OK and will work but will look better with a switch as it's likely that this check is extended in the future with more type checks.

Reply:
This decorator was giving problems again - same as when adding it to Organisation. See comment in api doc about switching to using a dedicated type for this...so think this class will go.